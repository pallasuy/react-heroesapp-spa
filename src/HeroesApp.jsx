import { AppRouter } from "./router/AppRouter"
import { AuthProvider } from "./auth"

export const HeroesApp = () => {
    return (
        <>
            <AuthProvider>
                <h1>HeroesApp</h1>
                <AppRouter />
            </AuthProvider>

        </>

    )
}
