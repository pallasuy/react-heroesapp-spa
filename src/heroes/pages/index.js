export * from './DC'
export * from './Hero'
export * from './Marvel'
export * from './Search'