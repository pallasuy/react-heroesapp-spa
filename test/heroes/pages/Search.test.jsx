

import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import { Search } from '../../../src/heroes';

describe('Pruebas en <Search />', () => {

    test('Debe de mostrar correctamente con valores por defecto', () => {
        const { container } = render(
            <MemoryRouter>
                <Search />
            </MemoryRouter>
        );

        //screen.debug();
        expect(container).toMatchSnapshot();

    })

    test('Debe de mostrar a batman y el input con el valor del queryString ', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Search />
            </MemoryRouter>
        );
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');

        const img = screen.getByRole('img');
        expect(img.src).toContain('/heroes/dc-batman.webp');
        //screen.debug();
        const alertDanger = screen.getByLabelText('alert-danger');
        expect(alertDanger.style.display).toBe('none');
       

    })

})