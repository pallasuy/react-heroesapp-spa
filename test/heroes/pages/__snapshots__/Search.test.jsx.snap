// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Search /> Debe de mostrar correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-4"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-8"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
         No hero with 
        <b>
           
        </b>
         
      </div>
    </div>
  </div>
</div>
`;
